src/main/scala/viper/gobra/frontend/ParseTreeTranslator.scala

```diff
@@ -1851,14 +1851,24 @@ class ParseTreeTranslator(pom: PositionManager, source: Source, specOnly : Boole
     * {@link #visitChildren} on {@code ctx}.</p>
     */
   override def visitTerminationMeasure(ctx: TerminationMeasureContext): PTerminationMeasure = {
+    // when parsing imported packages (`specOnly` will be true), we trust the annotations of imported members
+    // therefore, e.g. bodies of non-pure functions & methods will be ignored. However, the body of pure
+    // functions & methods is retained. To avoid that Viper will generate proof obligations for checking termination
+    // of imported pure functions & methods, we simply translate tuple termination measures into wildcard measures (for
+    // a given condition). Therefore, no proof obligations will be created for checking termination of imported members.
+    // This is technically not necessary for imported non-pure functions & methods because they are abstract
+    // anyway but we apply the same transformation for simplicity as we do not have to perform a case distinction on
+    // whether the termination measure is part of a pure or non-pure member.
```

Comment:
I wonder what happens if you import an interface whose methods have a termination measure. Do we change their termination measures to a wildcard then? If so, this sounds unsound to me, as one can still implement that interface in the importing package

Reply:
Very good point and you're absolutely right!
I've now adapted the approach to only adapt termination measures for method and function declarations. Since detecting this context within the parser seemed too hacky to me, I've added another parse tree postprocessor

Reply:
I have implemented your suggestion. However, I'm still turning decreases tuples into wildcards because Viper also checks whether postconditions terminate

Reply:
I would be carefull with that rewrite. If the function is recursive, then `body` would contain a call to itself which I vaguely remember to be unsound in Viper right now. Maybe ask Marco, Gaurav, or Thibault there.

Reply:
(As you might saw) I asked people, and I misremembered the problem: The problem are matching loops. For instance, the following program verifies:

```
function bar(x: Int): Int 
  ensures result == (x > 0 ? x + bar(x-1) : 0)

method test() {
  assert bar(98) == 4851
}
```

Therefore, this part of the transformation should be removed.

Reply:
FYI there is some global bound on the unrollings and 98 is the most it can do. The following programs does not verify because of that (which makes a nice riddle):

```
function bar1(x: Int): Int 
  ensures result == (x > 0 ? x + bar1(x-1) : 0)

function bar2(x: Int): Int 
  ensures result == (x > 0 ? x + bar2(x-1) : 0)

method test() {
  assert bar1(98) == 4851
  assert bar2(1) == 1
}
```